### ISSUE TITLE
Path rendering issues

### ISSUE THREAD

**Issue by stashymane** [hervegirod/fxsvgimage#30]
Using ionicons (only with the fill attribute added to the original SVG), some icons with paths are rendered very odd.
At least from what I have used so far, this includes the `person-circle` and `settings` icons.

![image](https://user-images.githubusercontent.com/4071883/173394940-e6c22852-7d8c-403a-a9d8-f9b31e805db1.png)

The exact SVG causing this:

```
<svg xmlns="http://www.w3.org/2000/svg" class="ionicon" viewBox="0 0 512 512" fill="#fff">
    <title>Person Circle</title>
    <path d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm-50.22 116.82C218.45 151.39 236.28 144 256 144s37.39 7.44 50.11 20.94c12.89 13.68 19.16 32.06 17.68 51.82C320.83 256 290.43 288 256 288s-64.89-32-67.79-71.25c-1.47-19.92 4.79-38.36 17.57-51.93zM256 432a175.49 175.49 0 01-126-53.22 122.91 122.91 0 0135.14-33.44C190.63 329 222.89 320 256 320s65.37 9 90.83 25.34A122.87 122.87 0 01382 378.78 175.45 175.45 0 01256 432z"/>
</svg>
```

**Comment by hervegirod** [comment]
I don't have the same result (I have nothing in my case), but I think that it is coming from the fact that I am relying too much on SVGPath in JavaFX (I am still not feeding JavaFX path the exact content of the path in the svtg file, because I am taking care for example of the units). 

I see that SVGPath does not handle correctly paths with several shapes in them, which is the case case in this one and a very simple one with 2 rectangles in one path  I created to test the behavior of my library.

I will need to create more than one SVGPah if there are more than one M or m in the path. Not very complex I think because as I said I am already parsing the content of the path and not feed the content directly to the SVGPath. Can you confirm that the other examples which are not working correctly also have more than one m or M?

**Comment by stashymane** [comment]
All of the ones that have broken rendering so far do have multiple shapes (M/m), however not all SVGs with multiple shapes are broken (for example - [the add icon](https://unpkg.com/ionicons@5.5.2/dist/svg/add-outline.svg)). The only difference I can see between these few examples are that the add icon only has outlines, while the rest have fills.

**Comment by hervegirod** [comment]
Should be fixed as of next 1.1 version
